libcentrifugo/engine/engineredis/hash_test.go

```diff
@@ -0,0 +1,16 @@
+package engineredis
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestHash(t *testing.T) {
+	chans := []string{"chan1", "chan2", "chan3", "chan4", "chan5", "chan6", "chan7", "chan8"}
+	for _, ch := range chans {
+		bucket := Hash(ch, 2)
+		assert.True(t, bucket >= 0)
+		assert.True(t, bucket < 2)
+	}
+}
```

Comment:
FWIW I can't see a defect in the implementation compared with the paper other than the redundant hash as noted. But I'm never confident of code that I can't at least see passing basic sanity checks!